homeassistant/components/unifiprotect/number.py

```diff
@@ -0,0 +1,191 @@
+        entity_category=EntityCategory.CONFIG,
+        ufp_min=0,
+        ufp_max=100,
+        ufp_step=1,
+        ufp_required_field="feature_flags.can_optical_zoom",
+        ufp_value="isp_settings.zoom_position",
+        ufp_set_function="set_camera_zoom",
+    ),
+    ProtectNumberEntityDescription(
+        key=_KEY_CHIME,
+        name="Chime Duration",
+        icon="mdi:camera-timer",
+        entity_category=EntityCategory.CONFIG,
+        ufp_min=0,
+        ufp_max=10000,
+        ufp_step=100,
+        ufp_required_field="feature_flags.has_chime",
+        ufp_value="chime_duration",
+        ufp_set_function="set_chime_duration",
+    ),
+)
+
+LIGHT_NUMBERS: tuple[ProtectNumberEntityDescription, ...] = (
+    ProtectNumberEntityDescription(
+        key=_KEY_SENSITIVITY,
+        name="Motion Sensitivity",
+        icon="mdi:walk",
+        entity_category=EntityCategory.CONFIG,
+        ufp_min=0,
+        ufp_max=100,
+        ufp_step=1,
+        ufp_required_field=None,
+        ufp_value="light_device_settings.pir_sensitivity",
+        ufp_set_function="set_sensitivity",
+    ),
+    ProtectNumberEntityDescription(
+        key=_KEY_DURATION,
+        name="Auto-shutoff Duration",
+        icon="mdi:camera-timer",
+        entity_category=EntityCategory.CONFIG,
+        ufp_min=15,
+        ufp_max=900,
+        ufp_step=15,
+        ufp_required_field=None,
+        ufp_value="light_device_settings.pir_duration",
+        ufp_set_function="set_duration",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up number entities for UniFi Protect integration."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[ProtectDeviceEntity] = async_all_device_entities(
+        data,
+        ProtectNumbers,
+        camera_descs=CAMERA_NUMBERS,
+        light_descs=LIGHT_NUMBERS,
+    )
+
+    async_add_entities(entities)
+
+
+class ProtectNumbers(ProtectDeviceEntity, NumberEntity):
+    """A UniFi Protect Number Entity."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        device: Camera | Light,
+        description: ProtectNumberEntityDescription,
+    ) -> None:
+        """Initialize the Number Entities."""
+        self.device: Camera | Light = device
+        self.entity_description: ProtectNumberEntityDescription = description
+        super().__init__(data)
+        self._attr_max_value = self.entity_description.ufp_max
+        self._attr_min_value = self.entity_description.ufp_min
+        self._attr_step = self.entity_description.ufp_step
+
+    @callback
+    def _async_update_device_from_protect(self) -> None:
+        super()._async_update_device_from_protect()
+
+        assert self.entity_description.ufp_value is not None
+
+        value: float | timedelta = get_nested_attr(
+            self.device, self.entity_description.ufp_value
+        )
```

Comment:
Wouldn't it make more sense to put a lambda into the entity description instead?

Reply:
Lambdas never really made sense before we did all of the re-factoring to make it all work as is now. I think lambdas definitely would be a great next refactor pass. Since switch is already merged in, I think I would want to do this as a separate follow up PR and then I can apply the new refactoring to select/sensor/binary sensor as I add them.

Reply:
It would be very much isolated in this platform, so why not do it right here and later refactor the others?

Reply:
The mixin for the `ufp_value` is shared with the switch platform.

Reply:
Please do the refactor in a PR that **only** has the lambda change after all the `ufp_value` PRs have merged.  I can then do a manual re-test of everything in one pass since I've already tested everything for a few hours with `ufp_value`

Reply:
I like that idea. Then we do not need to do a bunch of re-testing here or for the HACS integration

Reply:
I propose the following order, mostly because its the most logical for testing on my part.

- `number` (this PR)
- `binary_sensor` & docs (pre-refactor)
- `sensor` & docs (pre-refactor)
- `select` & docs (pre-refactor)
- lambda refactor
- docs refactor for entities
- `views` & docs (post-refactor)
- `services` & docs (post-refactor)